resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -145,6 +157,13 @@ private[spark] class ExecutorPodsAllocator(
       logDebug(s"Pod allocation status: $currentRunningCount running, " +
         s"${currentPendingExecutors.size} pending, " +
         s"${newlyCreatedExecutors.size} unacknowledged.")
+
+      val numFailedExecutors =
```

Comment:
I don't know about this necessarily being the best way to track the number of failed executors. In dynamic allocation it seems like if we scaled up and then down a few times this would give us a incorrect reading. Or is that taken care of elsewhere?

Reply:
Holden is right. To track failures you'll have to look at `ExecutorPodsLifecycleManager` instead.

Reply:
Oh, I will reconsider the acquisition of numFailedExecutors.